storey/timescaledb_target.py

```diff
@@ -0,0 +1,322 @@
+
+    def _test_connection_sync(self) -> None:
+        """Test database connection synchronously during initialization.
+
+        This method validates that the TimescaleDB instance is accessible and properly configured
+        before any events are processed. It performs the following checks:
+        1. Basic connectivity to PostgreSQL/TimescaleDB
+        2. Table existence validation (if table name is provided)
+        3. TimescaleDB extension availability (optional check)
+
+        Raises:
+            ConnectionError: If connection fails, table doesn't exist, or other database issues occur
+        """
+        try:
+            # Get or create event loop for sync context
+            try:
+                loop = asyncio.get_event_loop()
+            except RuntimeError:
+                # No event loop exists, create one
+                loop = asyncio.new_event_loop()
+                asyncio.set_event_loop(loop)
+
+            # Test connection synchronously
+            loop.run_until_complete(self._test_connection_async())
+
+        except Exception as e:
+            raise ConnectionError(f"Failed to connect to TimescaleDB: {e}") from e
+
+    async def _test_connection_async(self) -> None:
+        """Async helper for connection testing.
+
+        Creates a temporary connection to validate database accessibility and configuration.
+        This connection is separate from the main connection pool and is closed immediately
+        after testing.
+        """
+        # Create a temporary connection to test
+        conn = await asyncpg.connect(dsn=self._dsn)
+
+        try:
+            # Test basic connectivity
+            await conn.execute("SELECT 1")
+
+            # Check if TimescaleDB extension is installed
+            result = await conn.fetchrow("SELECT extversion FROM pg_extension WHERE extname = 'timescaledb';")
+            if not result:
+                raise ConnectionError("TimescaleDB extension is not installed")
+
+            # Test if table exists (if specified)
+            if self._table:
+                table_exists = await conn.fetchrow(
+                    "SELECT 1 FROM information_schema.tables WHERE table_name = $1 LIMIT 1", self._table
+                )
+                if not table_exists:
+                    raise ConnectionError(f"Table '{self._table}' does not exist")
+
+        finally:
+            await conn.close()
+
+    async def _create_pool_async(self) -> asyncpg.Pool:
+        """Create asyncpg connection pool asynchronously.
+
+        Establishes a connection pool with the specified DSN and connection limits.
+        The pool is configured for optimal performance with TimescaleDB's time-series workloads.
+
+        Returns:
+            asyncpg.Pool: Configured connection pool ready for use
+        """
+        return await asyncpg.create_pool(
+            dsn=self._dsn,
+            min_size=self._min_connections,
+            max_size=self._max_connections,
```

Comment:
I'm pretty sure under the current implementation, there can only be one connection in use at every given time.

Reply:
Set to _min_connections and _max_connections by default to 1

Reply:
The parameter is misleading though, if the implementation doesn't actually support it.